core/src/components/router/router.tsx

```diff
@@ -132,13 +133,14 @@ export class Router implements ComponentInterface {
     console.debug('[ion-router] URL pushed -> updating nav', url, direction);
 
     let path = parsePath(url);
-    let queryString = url.split('?')[1];
+    let queryString = url.substring(url.indexOf('?') + 1);
```

Comment:
I think this is wrong is `url.indexOf('?') == -1`

Reply:
What about adding a `let targetUrl` which would be equal to `url` or `redirect` and call `parsePath` and compute the `queryString` only once before `setPath` ?

Reply:
If `indexOf` returns `-1`, then it would be `url.substring(0)` which is just `url`.

Reply:
> If `indexOf` returns `-1`, then it would be `url.substring(0)` which is just `url`.

Yeah I think this is a problem. It should be empty in this case, no ?

Reply:
Also, would it make sense to propagate fragments ? `/route#fragment`

Reply:
Oh hah I got my variables mixed up. Yes it should just be `undefined`.

Reply:
Thinking more about it handling fragments is not that simple... as they could be routes (if `use-hash=true`).

But then maybe there is a problem with the handling of query string, what if `site.com?dark=1#/route` ?